File: essentials/18_db.py
"""
===========================================
PYTHON TUTORIAL 18
Databases
===========================================

pip install mysql-connector-python

- připojení k databázi
- SELECT dotaz
- výpis výsledků
"""

import mysql.connector # import knihovny pro práci s MySQL databází

print("Python Tutorial 18 - Práce s databází\n")

try:
    conn = mysql.connector.connect( # připojení k databázi
        host="localhost",
        user="root",
        password="",
        database="store"
    )

    if conn.is_connected(): # kontrola, zda je připojení úspěšné
        print("Připojeno k databázi\n")

except mysql.connector.Error as e: # zachycení chyby připojení a výpis chybové zprávy
    print(f"Chyba připojení: {e}")
    exit() # ukončení skriptu, pokud se připojení nezdařilo

 # vytvoření kurzoru pro provádění SQL dotazů, dictionary=True umožňuje přístup k výsledkům jako k slovníku
cursor = conn.cursor(dictionary=True)

sql_select = "SELECT * FROM warehouse;"
# provedení SQL dotazu, výsledky jsou nyní v kurzoru
cursor.execute(sql_select) 

# načtení všech výsledků do proměnné rows, každý řádek je slovník díky dictionary=True
rows = cursor.fetchall() 

# kontrola, zda byly načteny nějaké řádky, a výpis dat
if rows:
    print("Seznam skladů:\n")

    # iterace přes načtené řádky a výpis informací o každém skladu
    for row in rows:
        print(f"ID: {row['id']}")
        print(f"Jméno: {row['name']}")
        print(f"Ulice: {row['street']}")
        print(f"Město: {row['city']}")
        print(f"PSČ: {row['postal_code']}")
        print(f"Země: {row['country']}\n")

else:
    print("Žádná data v tabulce")

cursor.close() # uzavření kurzoru, uvolnění prostředků
conn.close() # uzavření připojení k databázi, uvolnění prostředků

print("Připojení uzavřeno")

File: essentials/18_db.php
<?php
/*
===========================================
PHP TUTORIAL 18
Databases
===========================================

Níže je kompletní jednoduchý skript, který:
- se připojí k databázi
- načte data (SELECT)
- vypíše výsledky

---
*/

echo "<h1>PHP Tutorial 18 - Práce s databází (MySQLi OOP)</h1>";

$host = "localhost";
$user = "root";
$password = "";
$database = "store";

$conn = new mysqli($host, $user, $password, $database);

if ($conn->connect_error) {
    die("Chyba připojení: " . $conn->connect_error);
}

echo "Připojeno k databázi\n\n";

$sqlSelect = "SELECT * FROM warehouse;";
$result = $conn->query($sqlSelect);

if ($result->num_rows > 0) {

    echo "<strong>Seznam skladů:</strong>\n\n";

    while ($row = $result->fetch_assoc()) {
        echo "ID: " . $row["id"] . "\n";
        echo "Jméno: " . $row["name"] . "\n";
        echo "Ulice: " . $row["street"] . "\n";
        echo "Město: " . $row["city"] . "\n";
        echo "PSČ: " . $row["postal_code"] . "\n";
        echo "Země: " . $row["country"] . "\n\n";
    }

} else {
    echo "Žádná data v tabulce";
}

$conn->close();

echo "Připojení uzavřeno";


